client/service/src/builder.rs

```diff
@@ -344,6 +344,8 @@ pub struct SpawnTasksParams<'a, TBl: BlockT, TCl, TExPool, TRpc, Backend> {
 	pub system_rpc_tx: TracingUnboundedSender<sc_rpc::system::Request<TBl>>,
 	/// Telemetry instance for this node.
 	pub telemetry: Option<&'a mut Telemetry>,
+	/// Custom subscription generator for JSON-RPC subscriptions.
+	pub rpc_id_provider: TRpcId,
```

Comment:
Could this be `Option<TRpcId>` like `telemetry` is? Or is this the wrong place for setting defaults? If possible I think we should make the default be numeric IDs and let the teams that need it override to strings but I'm not sure what the best way to achieve that is. Not possible?

Reply:
I thought I made a comment about that but must have disappeared.

The reason is https://docs.rs/jsonrpsee-ws-server/latest/jsonrpsee_ws_server/struct.WsServerBuilder.html#method.set_id_provider, which requires static dispatch and you can't pass in the concrete type if it's `None`.

Perhaps we should change the API in jsonrpsee but not possible AFAIU.